rfd/0170-package-distribution-v3.md

```diff
@@ -0,0 +1,450 @@
+customer-facing download requests would need to flow through a service that we
+are responsible for keeping online and available, failing this requirement.
+
+##### Artifactory
+
+JFrog Artifactory meets all of our requirements, and is the *only* solution on
+the market today that does so. In addition to meeting our requirements, we would
+see several other benefits with this solution:
+
+* Artifactory has support for several other repository types that we currently
+  have. By (eventually) using Artifactory for these, we could get rid of a lot
+  of tooling and infrastructure. Here's a list of repos types that we could
+  shift to this solution:
+  * Helm
+    * Currently requires custom tooling for our release pipelines
+  * Terraform (Nexus Repository requires a third-party plugin)
+    * Currently requires extensive custom tooling for our release pipelines, as
+      well as tooling for disaster recovery
+  * Release server/raw binary artifacts
+    * Currently requires a large amount of tooling in a monolithic service that
+      struggles to support our current needs
+* There is support for pull-through caches and/or mirroring for all major
+  toolchains that we use in our product today. By using Artifactory (eventually)
+  as a pull-through cache, we could scan all dependencies for vulnerabilities,
+  malicious code, and changes in licensing prior to download. This would not
+  only increase stability of our release pipelines, but also improve the
+  security of our developer endpoints. here's a list of package types that are
+  supported here:
+  * OS packages (debs, rpms)
+    * We don't pin these at all in our release pipelines, meaning we currently
+      blindly trust the upstream sources
+  * Go modules
+    * We pin these currently but do not actively scan these for new/unreported
+      vulnerabilities
+  * Rust creates
+    * We pin these currently but do not actively scan these for new/unreported
+      vulnerabilities
+  * NPM packages
+    * We have constant failures due to issues with upstream sources
+    * NPM has a history of being used for distributing malware, and JFrog's
+      solution [has a history of detecting
+      it](https://jfrog.com/blog/malware-civil-war-malicious-npm-packages-targeting-malware-authors/)
+  * Container images
+    * We don't often hit rate limiting here, but only because we pass around
+      shared dockerhub credentials, which this solution could eliminate
+    * We scan our built images for security issues, but not the images we pull
+      in to build our releases (which have huge amounts of access to sensitive
+      data)
+* We could replace several of our current security tools **and vendors** with
+  tools that come included with a self-hosted JFrog license. This aligns with
+  our high-level objective of reducing our dependence on vendor services,
+  despite adding a new vendor. This includes:
+  * Orca (SaaS service)
```

Comment:
Orca we're already removing and replacing with Wiz so it's not relevant here.

Reply:
Have we already signed with Wiz? If not, would there be value in using JFrog's solution instead of Wiz, so that we don't need another vendor?

Reply:
This is irrelevant to this RFD, Wiz implementation is already underway. Let's just remove it from scope altogether here.